### PR TITLE
fix(prometheus_exporter sink): Avoid extra work in bad requests

### PR DESCRIPTION
**Pull request by bruceg**
As written, the cost of counting event byte size is taken even if the request is unauthorized or hits the wrong path. This scopes the count operation down into the path where the request was successful. As a side effect, it also prevents the sink from emitting the events sent counter when no events were sent.

This is another piece found while taking a look at an relatively unrelated exporter issue.

### CODE REVIEW

File: src/sinks/prometheus/exporter.rs [hunk 1]
```diff
@@ -374,28 +374,37 @@ fn handle(
...
+        (false, _, _) => {
+            *response.status_mut() = StatusCode::UNAUTHORIZED;
+            response.headers_mut().insert(
+                http::header::WWW_AUTHENTICATE,
+                HeaderValue::from_static("Basic, Bearer"),
+            );
+        }
+
+        (true, &Method::GET, "/metrics") => {
+            let metrics = metrics.read().unwrap();
```
Comment (jszwedko) [inline src/sinks/prometheus/exporter.rs:392]: `expect` instead?

**Review by jszwedko — APPROVED** [review]
I think there is a question of whether we even want this metric in this sink, but 👍 since the behavior is definitely more correct now in the error case.

**Review by tobz — APPROVED** [review]
Makes sense to me. 👍🏻 

I agree With Jesse's nit about using `expect` instead of `unwrap`, though... but non-blocking.

File: src/sinks/prometheus/exporter.rs (hunk 1 above)
Comment (bruceg) [reply src/sinks/prometheus/exporter.rs:392]: That was copied from the existing code, but I agree an expect would improve the situation.

**Pull request merged**